src/Disks/IDiskRemote.cpp

```diff
@@ -359,7 +359,16 @@ void IDiskRemote::getRemotePathsRecursive(const String & local_path, std::vector
 {
     if (metadata_disk->isFile(local_path))
     {
-        paths_map.emplace_back(local_path, getRemotePaths(local_path));
+        try
+        {
+            paths_map.emplace_back(local_path, getRemotePaths(local_path));
+        }
+        catch (const Exception & e)
+        {
+            if (e.code() == ErrorCodes::FILE_DOESNT_EXIST)
+                return;
+            throw;
+        }
```

Comment:
Comment is needed. Btw, what will happen if `isFile` returned `true`, but another thread removed the file and created directory with the same path? Or if `isFile` returned `false`, because it was directory, but the directory was removed concurrently?

Reply:
> Comment is needed.

done

Reply:
> I believe this case is impossible, that there was a file but then a directory on its place.

`IDisk` interface allows to remove file and create directory with the same name. Yes, seems like there is no code in ClickHouse that actually does it, but who knows... 
However, I think we can just ignore this case for now.